Relation-SPA/src/app/_services/relation.service.ts

```diff
@@ -8,12 +12,16 @@ import { HttpClient } from '@angular/common/http';
 export class RelationService {
   constructor(private _httpClient: HttpClient) { }
 
-
-
-  public GetRelations() {
-   return this._httpClient.get('http://localhost:5000/api/Relations');
+  public GetRelations(id?: string): Observable<Relation[]> {
+    let params = new HttpParams();
+    if (id != null) {
+      params = params.append('categoryId', id);
+    }
+    return this._httpClient.get<Relation[]>('http://localhost:5000/api/Relations', { params });
```

Comment:
Do not hardcode "http://localhost:5000/". Put "http://localhost:5000/" into variable in one place so that you can use this variable across all the project.

Reply:
"http://localhost:5000/" was successfully encapsulated in environment.ts file for proper re usage in services;